src/evict.c

```diff
@@ -520,11 +520,15 @@ int performEvictions(void) {
     int slaves = listLength(server.slaves);
     int result = EVICT_FAIL;
 
-    if (getMaxmemoryState(&mem_reported,NULL,&mem_tofree,NULL) == C_OK)
-        return EVICT_OK;
+    if (getMaxmemoryState(&mem_reported,NULL,&mem_tofree,NULL) == C_OK) {
```

Comment:
@oranagra I wander whether line 513 is needed.

Reply:
what do you mean by "needed"? are you suggesting to delete it? or saying that in that early return we don't need to update the metrics?

if that's the second case, then i think you're right. that early return is as if we didn't call this function.
but better add some comment there that says that we don't go to `update_metrics` on purpose because it's as if we didn't call this function. (it's a fake EVICT_OK).

p.s. that exact line used to be implemented in a wrapper function in the past 8-)

Reply:
I mean second case. At replica, if we change `server.repl_slave_ignore_maxmemory`, then maybe these metrics can't be updated forever. But this is rarely.